lxd/device/gpu.go

```diff
@@ -128,7 +195,38 @@ func (d *gpu) startContainer() (*deviceConfig.RunConfig, error) {
 		if gpu.DRM != nil && (d.config["id"] == "" || fmt.Sprintf("%d", gpu.DRM.ID) == d.config["id"]) {
 			found = true
 
-			if gpu.DRM.CardName != "" && gpu.DRM.CardDevice != "" && shared.PathExists(filepath.Join(gpuDRIDevPath, gpu.DRM.CardName)) {
+			if d.config["mdev"] != "" {
+				v := d.volatileGet()
+
+				// Get iommu group
+				vgpuPath := fmt.Sprintf("/sys/bus/pci/devices/%s/%s/iommu_group", gpu.PCIAddress, v["vgpu.uuid"])
+
+				link, err := os.Readlink(vgpuPath)
+				if err != nil {
+					return nil, err
+				}
+
+				iommuGroup := filepath.Base(link)
+				path := filepath.Join(gpuVFIODevPath, iommuGroup)
+
+				// Get major and minor numbers
+				var stat unix.Stat_t
+
+				err = unix.Stat(path, &stat)
+				if err != nil {
+					return nil, err
+				}
+
+				major := uint32(stat.Rdev / 256)
+				minor := uint32(stat.Rdev % 256)
+
+				err = unixDeviceSetupCharNum(d.state, d.inst.DevicesPath(), "unix", d.name, d.config, major, minor, path, false, &runConf)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			if d.config["mdev"] == "" && gpu.DRM.CardName != "" && gpu.DRM.CardDevice != "" && shared.PathExists(filepath.Join(gpuDRIDevPath, gpu.DRM.CardName)) {
```

Comment:
@monstermunchkin could this be an `else if` here rather than a separate if block?